apis/v1alpha1/gateway_types.go

```diff
@@ -709,38 +711,140 @@ type ListenerStatus struct {
 // field.
 type ListenerConditionType string
 
+// ListenerConditionReason defines the set of reasons that explain
+// why a particular Listener condition type has been raised.
+type ListenerConditionReason string
+
 const (
-	// ConditionInvalidListener is a generic condition that is a catch all for
-	// unsupported configurations that do not match a more specific condition.
-	// Implementers should try to use a more specific condition instead of this
-	// one to give users and automation more information.
-	ConditionInvalidListener ListenerConditionType = "InvalidListener"
-
-	// ConditionListenerNotReady indicates the listener is not ready.
-	ConditionListenerNotReady ListenerConditionType = "ListenerNotReady"
-
-	// ConditionPortConflict indicates that two or more Listeners with
-	// the same port were bound to this gateway and they could not be
-	// collapsed into a single configuration.
-	ConditionPortConflict ListenerConditionType = "PortConflict"
-
-	// ConditionInvalidCertificateRef indicates the certificate reference of the
-	// listener's TLS configuration is invalid.
-	ConditionInvalidCertificateRef ListenerConditionType = "InvalidCertificateRef"
-
-	// ConditionRoutesNotReady indicates that at least one of the specified
-	// routes is not ready.
-	ConditionRoutesNotReady ListenerConditionType = "RoutesNotReady"
-
-	// ConditionInvalidRoutes indicates that at least one of the specified
-	// routes is invalid.
-	ConditionInvalidRoutes ListenerConditionType = "InvalidRoutes"
-
-	// ConditionForbiddenRoutesForClass indicates that at least one of the
-	// routes is in a namespace forbidden by the GatewayClass.
-	ConditionForbiddenRoutesForClass ListenerConditionType = "ForbiddenRoutesForClass"
-
-	// ConditionUnsupportedProtocol indicates that an invalid
-	// or unsupported protocol type was requested.
-	ConditionUnsupportedProtocol ListenerConditionType = "UnsupportedProtocol"
+	// ListenerConditionConflicted indicates that the controller
+	// was unable to resolve conflicting specification requirements
+	// for this Listener. If a Listener is conflicted, its network
+	// port should not be configured on any network elements.
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "HostnameConflict"
+	// * "ProtocolConflict"
+	// * "RouteConflict"
+	//
+	// Controllers may raise this condition with other reasons,
+	// but should prefer to use the reasons listed above to improve
+	// interoperability.
+	ListenerConditionConflicted ListenerConditionType = "Conflicted"
+
+	// ListenerReasonHostnameConflict is used when the Listener
+	// violates the Hostname match constraints for that allow
+	// collapsing Listeners. For example, this reason would be
```

Comment:
I'm having difficulty parsing "for that allow".  Is this saying that there is a set of conditions that necessitate collapsing listeners and a set of conditions that prohibit collapsing listeners, and `HostnameConflict` indicates that both sets of conditions were met for the same group of listeners?

Reply:
Good catch! This should be either "violates the Hostname match constraints that allow collapsing" or "violates the Hostname match constraints for collapsing". Looks like I got stuck half-way :)